charts/tests/ut/cp_statefulset.bats

```diff
@@ -0,0 +1,547 @@
+}
+
+@test "Control Plane StatefulSet: Annotations, global annotations and commonAnnotations can be set" {
+    template \
+        --set 'global.annotations.foo=bar' \
+        --set 'commonAnnotations.baz=baa' \
+        --set 'annotations.qux=quux'
+    assert_query_equal '.metadata.annotations.foo' "bar"
+    assert_query_equal '.metadata.annotations.baz' "baa"
+    assert_query_equal '.metadata.annotations.qux' "quux"
+}
+
+@test "Control Plane StatefulSet: Recommended labels are set" {
+    template
+    assert_query_equal '.metadata.labels."app.kubernetes.io/name"' "xrd-control-plane"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/instance"' "release-name"
+    assert_query_equal '.metadata.labels."app.kubernetes.io/managed-by"' "Helm"
+    assert_query '.metadata.labels | has("app.kubernetes.io/version")'
+    assert_query '.metadata.labels | has("helm.sh/chart")'
+}
+
+@test "Control Plane StatefulSet: Global labels and commonLabels can be set" {
+    template --set 'global.labels.foo=bar' --set 'commonLabels.baz=baa'
+    assert_query_equal '.metadata.labels.foo' "bar"
+    assert_query_equal '.metadata.labels.baz' "baa"
+}
+
+@test "Control Plane StatefulSet: Labels can be added" {
+    template --set 'labels.foo=bar'
+    assert_query_equal '.metadata.labels.foo' "bar"
+}
+
+@test "Control Plane StatefulSet: Replicas is one" {
+    template
+    assert_query_equal '.spec.replicas' "1"
+}
+
+@test "Control Plane StatefulSet: serviceName is the name of the StatefulSet" {
+    template
+    assert_fields_equal '.spec.serviceName' '.metadata.name'
+}
+
+@test "Control Plane StatefulSet: Selector labels are set" {
+    template
+    assert_multiline_query_equal '.spec.selector.matchLabels' \
+        "app.kubernetes.io/name: xrd-control-plane\napp.kubernetes.io/instance: release-name"
+}
+
+@test "Control Plane StatefulSet: No .spec.template annotations are set by default" {
+    template
+    assert_query '.spec.template.metadata.annotations | not'
+}
+
+@test "Control Plane StatefulSet: .spec.template global annotations and commonAnnotations can be set" {
+    template --set 'global.annotations.foo=bar' --set 'commonAnnotations.baz=baa'
+    assert_query_equal '.spec.template.metadata.annotations.foo' "bar"
+    assert_query_equal '.spec.template.metadata.annotations.baz' "baa"
+}
+
+@test "Control Plane StatefulSet: .spec.template annotations are added for multus interfaces" {
+    template --set-json 'interfaces=[{"type": "multus"}]'
+    assert_multiline_query_equal '.spec.template.metadata.annotations."k8s.v1.cni.cncf.io/networks"' \
+        "[\n  {\n    \"name\": \"release-name-xrd-control-plane-0\"\n  }\n]"
+}
+
+@test "Control Plane StatefulSet: .spec.template podAnnotations can be set" {
+    template --set 'podAnnotations.foo=bar'
+    assert_query_equal '.spec.template.metadata.annotations.foo' "bar"
+}
+
+@test "Control Plane StatefulSet: podNetworkAnnotations contain the desired information" {
```

Comment:
This is quite a weird feature tbh (at least seems odd for `attachmentConfig` to be specified per-interface) - I can check with Matt on this (no need to block this PR)